### DISCUSSION TITLE
Как разделить стратегии по интерфейсам?

### DISCUSSION THREAD

**Discussion by mikeS7** [bol-van/zapret#729] [Q&A] [answered]
Имею 2-х провайдеров, надо разделить стратегии в custom скриптах по интерфейсам.
Функция nft_fw_nfqws_post всегда добавляет список wan интерфейсов, можно полностью правило с нуля написать, но может есть другой способ?

**Comment by bol-van** [comment]
Если система не openwrt, а обычный linux, то там можно не задавать переменную IFACE_WAN в конфиге.
Тогда хелперы не будут добавлять `oifname @wanif`. И дальше использовать FW_EXTRA_PRE.
Если openwrt, то там зашито без возможности убрать

**Reply by bol-van** [reply, marked as answer]
Если openwrt, можно вот что сделать.
OPENWRT_WAN="wan1 wan2"
В FW_EXTRA_PRE добавить еще раз oifname if_wan1, oifname if_wan2

**Reply by mikeS7** [reply]
> Если openwrt, можно вот что сделать. OPENWRT_WAN="wan1 wan2" В FW_EXTRA_PRE добавить еще раз oifname if_wan1, oifname if_wan2

Допилил немного для ipv6, у меня ipv6 только на одном прове. Буду проверять

**Reply by bol-van** [reply]
Если так, то можно делать локальные переопределения DISABLE_IPV6=1 при вызове хелперов

**Reply by mikeS7** [reply]
> Если так, то можно делать локальные переопределения DISABLE_IPV6=1 при вызове хелперов

Именно так и сделал:
```
        local DISABLE_IPV4=1
        local inputIn="meta mark & 0x04000000 == 0x04000000"
        nft_fw_nfqws_post "$inputIn" "$inputIn" $QNUM_RT_GV

        DISABLE_IPV4=0
        local DISABLE_IPV6=1
        local FW_EXTRA_PRE="oifname eth0.2202"
        nft_fw_nfqws_post "$inputIn" "$inputIn" $QNUM_RT_GV
```
получил такие правила:
```
oifname "eth0.2202" oifname @wanif meta mark & 0x04000000 == 0x04000000 ip daddr != @nozapret meta mark set meta mark | 0x20000000 queue flags bypass to 65400
oifname @wanif6 meta mark & 0x04000000 == 0x04000000 ip6 daddr != @nozapret6 meta mark set meta mark | 0x20000000 queue flags bypass to 65400
```

**Reply by bol-van** [reply]
так оно на одну очередь переадресует $QNUM_RT_GV
в чем смысл данного скрипта ?
если разница только в стратегии по ipv4 и ipv6, то весь этот огород не нужен. есть --filter-l3

и редирект идет без всякого фильтра всего исходящего трафика. это тоже ненормально и создает мощную и бесполезную нагрузку на nfqws. разве что до этого марк ставится по каким-то правилам, учитывающим порты и прочее

**Reply by mikeS7** [reply]
> в чем смысл данного скрипта ?
Таких скриптов несколько, это виртуалка с несколькими "входящими" и "исходящими" интерфейсами. Настроены правила маршрутизации и маркируется входящий трафик в зависимости от интерфейса. Т.о. я просто настраиваю стратегию под вход-выход.
Это хозяйство стоит рядом с микротиком (на нем вся фильтрация осуществляется) и трафик только "нужный" идет, пока это больше для изучения возможностей.
